elternportal.js

```diff
@@ -477,10 +477,10 @@ async function readNoticeBoard(page, previousHashes) {
   const subjects = await page.$$eval('div.well h4', hh => hh.map(h => h.innerHTML));
   const currentContents = await page.$$eval('div.well h4 ~ p', pp => pp.map(p => p.outerHTML));
   const archivedContents =
-    await page.$$eval('div.well div.row ~ div.row p', pp => pp.map(p => p.outerHTML));
+    await page.$$eval('div.well div.row ~ div.row p:first-child', pp => pp.map(p => p.outerHTML));

```

Comment:
I'm not 100% sure, that this works for other portals as well.
And currently I've only one message in archive, so I cannot check, what happens if there are more than one.

Reply:
Now both messages are archived and I still get
```
2024-10-31 07:18:03 error: Error in processElternPortal:
Error: Found 2 subjects, but 3 contents
    at readNoticeBoard (/home/pptruser/elternportal.js:487:11)
```

The problem is, that the attachment is counted as an additional content. In order to reproduce, you would need a announcement with an attachment.

Maybe have a  fix that is more specific? Looking a the html tags, it's hard, so look for "Anhang"? (also ugly).

Alternatively, count outer divs `<div class="row arch collapse in" aria-expanded="true" style="">` - I see two of them in archive.
Unfortunately, I now only have 2 messages in archive, but no current one to test, if this breaks the current messages counting.
What do you think?

Reply:
Answering to myself: "..if this breaks the current messages counting.": It won't, as `archivedContents` shall only keep the count or archived messages. If we add `div.arch` to the selector, this should work.
So my proposal is `div.arch div.well div.row ~ div.row p:first-child`